Feature/filter listings

Doorsfan opened the pull request:
Integrated so that Filters apply on the startpage - By utilizing wildcard designations, i have managed to make a largely functional multi attribute driven filter mechanic.

We could apply this to account for amenities and other matters too, should we see fit.

Doorsfan committed:
Merge branch 'main' into feature/filterListings

Doorsfan committed:
Integrated Filtering into Startpage

Integrated so multiple factor filtering is integrated into the Startpage in bigger strokes - Maybe can consider some form of reset button or so - but see no real pressing matter on this

mattkocak reviewed (changes requested):
1. There's an error when sending a blank string to the backend for both the "number of guests" and "price" fields. Recreate this bug by searching for 4 guests and then searching again, but clearing this text box. The same goes for the price field.
2. It seems like the filter doesn't work if the date fields are left blank. Is it possible to simply ignore the date field in these cases so that only the selected filters apply?

Doorsfan committed:
Basic Filtering

After many hours of attempting to perform a deeper check in terms of Dates and filtering there of - I have to concede on the matter for now.

I have at least implemented on the basis of being able to filter on Location, Guests, Start Date, End Date and price.

The only problem is that i have not figured out how to compare dates in JPA, as i believe this would require a lot of date formatting to check "ranges" of dates in that sense.

I attempted to resolve this multiple times by splitting up the DB columns into Years, Days and Months and other matters.

None of which gave results that worked as intended.

Alas, i will stick with this version for now.

Doorsfan commented:
I have now updated so that there are default case handlings in terms of blanked values.

For many hours i tried to implement a range in terms of START DATE to END DATE, but this
proved very difficult in terms of both JPA and Java in general.

Alas, i have performed the changes you requested.
